beacon_node/network/src/service.rs

```diff
@@ -237,22 +229,21 @@ impl<T: BeaconChainTypes> NetworkService<T> {
             "Backfill is disabled. DO NOT RUN IN PRODUCTION"
         );
 
-        // try and construct UPnP port mappings if required.
-        if let Some(upnp_config) = crate::nat::UPnPConfig::from_config(config) {
-            let upnp_log = network_log.new(o!("service" => "UPnP"));
-            let upnp_network_send = network_senders.network_send();
-            if config.upnp_enabled {
-                executor.spawn_blocking(
-                    move || {
-                        crate::nat::construct_upnp_mappings(
-                            upnp_config,
-                            upnp_network_send,
-                            upnp_log,
-                        )
-                    },
-                    "UPnP",
-                );
-            }
+        if let (true, false, Some(v4)) = (
+            config.upnp_enabled,
+            config.disable_discovery,
+            config.listen_addrs().v4(),
+        ) {
+            let nw = network_log.clone();
+            let v4 = v4.clone();
+            tokio::spawn(async move {
```

Comment:
Should we use `executor.spawn` to spawn a future here, since we were using `executor.spawn_blocking` before?

Reply:
yeah, thanks for the catch Akihito! Updated